libtransmission/peer-mgr.cc

```diff
@@ -412,18 +231,16 @@ class tr_swarm
     {
         auto const lock = unique_lock();
 
-        auto* const atom = peer->atom;
-        TR_ASSERT(atom != nullptr);
-
-        atom->time = tr_time();
```

Comment:
Oh I see, you combined 2 timestamps into 1.

Reply:
Hmmm.... when I refactored the code it seemed like everything the `time` field was doing could be handled by one of the other timestamp fields, but this part might be a good argument for it. :thinking:

Reply:
Looks the intended purpose of setting `atom->time` on closing connections is to prevent reconecting immediately after disconnecting.

https://github.com/transmission/transmission/blob/2ec19e694d0ac9226808ac04305e3e90b2b7f1ce/libtransmission/peer-mgr.cc#L2257-L2260

It didn't work as intended because you won't be able to initiate an outgoing connection to a temporary socket anyway. 🤷🏻‍♂️
If you'd like to keep this functionality, I'd suggest leave this as-is for this PR. I will fix this in one go when I separate the pool of listening sockets and temporary sockets.

Reply:
Updated in 08ed17d; `tr_peer_info.connection_changed_at_` now does what `peer_atom.time` used to, but with a more descriptive name.